fscache.go

```diff
@@ -175,105 +176,51 @@ func (c *cache) Clean() error {
 }
 
 type cachedFile struct {
-	fs   FileSystem
-	name string
-	cnt  int64
-	grp  sync.WaitGroup
-	w    io.WriteCloser
-	b    *broadcaster
+	stream *stream.Stream
+	cnt    int64
 }
 
 func (c *cache) newFile(name string) (*cachedFile, error) {
-	f, err := c.fs.Create(name)
+	s, err := stream.NewStream(name, c.fs)
 	if err != nil {
 		return nil, err
 	}
 	cf := &cachedFile{
-		fs:   c.fs,
-		name: f.Name(),
-		w:    f,
-		b:    newBroadcaster(),
+		stream: s,
 	}
-	cf.grp.Add(1)
 	atomic.AddInt64(&cf.cnt, 1)
 	return cf, nil
 }
 
 func (c *cache) oldFile(name string) *cachedFile {
-	b := newBroadcaster()
-	b.Close()
-	return &cachedFile{
-		fs:   c.fs,
-		name: name,
-		b:    b,
+	//FIXME
+	s, _ := stream.NewStream(name, c.fs)
```

Comment:
Looking at the Travis tests, it's hanging during "stream.Remove()" which makes sense since stream.Remove() blocks until all handles to the stream are closed.

Try adding s.Close() right after you create the stream, that'll tell the stream that you're done writing (which is true since oldFile is only used on complete, closed files).